frontend/src/scenes/data-warehouse/editor/multitabEditorLogic.tsx

```diff
@@ -1155,37 +1300,94 @@ export const multitabEditorLogic = kea<multitabEditorLogicType>([
                 })
             }
         },
-        activeModelUri: (activeModelUri) => {
-            if (props.monaco) {
-                const _model = props.monaco.editor.getModel(activeModelUri.uri)
-                const val = _model?.getValue()
-                actions.setQueryInput(val ?? '')
-                if (activeModelUri.sourceQuery) {
-                    actions.setSourceQuery({
-                        ...activeModelUri.sourceQuery,
-                        source: {
-                            ...activeModelUri.sourceQuery.source,
-                            query: val ?? '',
-                        },
-                    })
-                } else {
-                    actions.setSourceQuery({
-                        kind: NodeKind.DataVisualizationNode,
-                        source: {
-                            kind: NodeKind.HogQLQuery,
-                            query: val ?? '',
-                        },
-                        display: ChartDisplayType.ActionsLineGraph,
+        activeModelUri: async (activeModelUri) => {
+            if (!activeModelUri || !props.monaco) {
+                return
+            }
+
+            const _model = props.monaco.editor.getModel(activeModelUri.uri)
+            const val = _model?.getValue()
+            actions.setQueryInput(val ?? '')
+
+            const path = activeModelUri.uri.path.split('/').pop()
+            let storedVariables = {}
+            if (path) {
+                const storedVariablesString = await getStorageItem(`${activeModelVariablesStateKey(props.key)}_${path}`)
```

Comment:
Why are we storing the variables state in another key? `activeModelVariablesStateKey` exists but is dead code in `master` - we dont want to be storing (and reading) another state for variables - variables live on the query node (`values.sourceQuery.source.variables`)